test/common/runtime/runtime_impl_test.cc

```diff
@@ -678,6 +678,29 @@ TEST_F(StaticLoaderImplTest, ProtoParsing) {
   EXPECT_EQ(2, store_.gauge("runtime.num_layers", Stats::Gauge::ImportMode::NeverImport).value());
 }
 
+TEST_F(StaticLoaderImplTest, InvalidNumerator) {
+  base_ = TestUtility::parseYaml<ProtobufWkt::Struct>(R"EOF(
+    invalid_numerator:
+      numerator: 111
+      denominator: HUNDRED
+  )EOF");
+  setup();
+
+  envoy::type::v3::FractionalPercent fractional_percent;
+
+  // Numerator > denominator -> exception
+  EXPECT_CALL(generator_, random()).WillOnce(Return(500000));
```

Comment:
Just pull this into the #ifndef and get rid of the #else block. That should be the last of it from my end.